tools/ark-plugins/internal/plugins/owner_reference.go

```diff
@@ -0,0 +1,80 @@
+package plugins
+
+import (
+	"github.com/pkg/errors"
+
+	"github.com/heptio/ark/pkg/apis/ark/v1"
+	"github.com/heptio/ark/pkg/restore"
+	"github.com/sirupsen/logrus"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	scApi "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1beta1"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/clientset/typed/servicecatalog/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	restclient "k8s.io/client-go/rest"
+)
+
+// SetOwnerReference is a plugin for ark to set new value of UID in metadata.ownerReference field
+// based on new values of other restored objects
+type SetOwnerReference struct {
+	Log logrus.FieldLogger
+}
+
+// AppliesTo return list of resource kinds which should be handled by this plugin
+func (p *SetOwnerReference) AppliesTo() (restore.ResourceSelector, error) {
+	return restore.ResourceSelector{
+		IncludedResources: []string{"secret"},
+	}, nil
+}
+
+// Execute contains main logic for plugin
+// nolint
+func (p *SetOwnerReference) Execute(item runtime.Unstructured, restore *v1.Restore) (runtime.Unstructured, error, error) {
+	metadata, err := meta.Accessor(item)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	scClient, err := p.inClusterServiceCatalogClient()
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "while creating ServiceCatalog client")
+	}
+
+	sb, err := scClient.ServiceBindings(metadata.GetNamespace()).Get(metadata.GetName(), metav1.GetOptions{})
+	if err != nil {
```

Comment:
imo it will more readable to have the switch statement
```go
	switch {
	case err == nil:
	case apierrors.IsNotFound(err):
       	// there's no servicebinding with such name so the secret should not have ownerReference set
		p.Log.Infof("Couldn't get SB %s: %v", metadata.GetName(), err)
		return item, nil, nil
	default:
		return item, err, nill
	}
```

Reply:
changed